File: examples/movies-complete/src/main/java/org/superbiz/injection/tx/Movie.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package org.superbiz.injection.tx;

import javax.persistence.Entity;

@Entity
public class Movie {

    private String director;
    private String title;
    private int year;

    public Movie() {
    }

    public Movie(String director, String title, int year) {
        this.director = director;
        this.title = title;
        this.year = year;
    }

    public String getDirector() {
        return director;
    }

    public void setDirector(String director) {
        this.director = director;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }


}

File: examples/rest-mp-jwt/src/main/webapp/app/js/model/movie.js
/**
 *
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

(function () {
    'use strict';

    var deps = ['lib/backbone'];
    define(deps, function () {
        var isString = function (obj) {
            return Object.prototype.toString.call(obj) === '[object String]';
        };
        return Backbone.Model.extend({
            urlRoot: window.ux.ROOT_URL + 'rest/movies',
            idAttribute: 'id',
            toJSON: function () {
                if (!!this.attributes.rating && isString(this.attributes.rating)) {
                    this.attributes.rating = parseInt(this.attributes.rating, 10);
                }
                if (!!this.attributes.year && isString(this.attributes.year)) {
                    this.attributes.year = parseInt(this.attributes.year, 10);
                }
                if (!!this.attributes.id && isString(this.attributes.id)) {
                    this.attributes.id = parseInt(this.attributes.id, 10);
                }
                return this.attributes;
            },
            defaults: {
                rating: 5,
                year: new Date().getFullYear()
            }
        });
    });
}());